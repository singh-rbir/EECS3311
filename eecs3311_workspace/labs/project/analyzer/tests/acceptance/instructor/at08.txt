-- An example program that's type correct.
add_class("A")
add_attribute("A", "i", "INTEGER")
add_command("A", "set_i", <<["x", "INTEGER"], ["y", "INTEGER"]>>)

add_assignment_instruction("A", "set_i", "i")

-- Now the RHS of an assignment `i = ?` in the command `set_i` in class `A` is being specified.

-- The intended RHS of assignment: ((1 + 2) * 3)
multiplication 
addition
int_value (1)
int_value (2)
int_value (3)

add_assignment_instruction("A", "set_i", "i")

-- Now again the RHS of an assignment `i = ?` in the command `set_i` in class `A` is being specified.

addition
add_call_chain(<<"i">>) -- referring to attribute `i`
subtraction
add_call_chain(<<"x">>) -- referring to parameter `x`
add_call_chain(<<"y">>) -- referring to parameter `y`

-- RHS fully specified, we can then type-check
type_check

generate_java_code
