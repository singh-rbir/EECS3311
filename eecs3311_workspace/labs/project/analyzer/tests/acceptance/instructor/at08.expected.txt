  Status: OK.
  Number of classes being specified: 0
->add_class("A")
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 0
      Number of queries: 0
      Number of commands: 0
->add_attribute("A","i","INTEGER")
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + i: INTEGER
      Number of queries: 0
      Number of commands: 0
->add_command("A","set_i",<<["x", "INTEGER"], ["y", "INTEGER"]>>)
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + i: INTEGER
      Number of queries: 0
      Number of commands: 1
        + set_i(INTEGER, INTEGER)
->add_assignment_instruction("A","set_i","i")
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + i: INTEGER
      Number of queries: 0
      Number of commands: 1
        + set_i(INTEGER, INTEGER)
  Routine currently being implemented: {A}.set_i
  Assignment being specified: i := ?
->multiplication
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + i: INTEGER
      Number of queries: 0
      Number of commands: 1
        + set_i(INTEGER, INTEGER)
  Routine currently being implemented: {A}.set_i
  Assignment being specified: i := (? * nil)
->addition
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + i: INTEGER
      Number of queries: 0
      Number of commands: 1
        + set_i(INTEGER, INTEGER)
  Routine currently being implemented: {A}.set_i
  Assignment being specified: i := ((? + nil) * nil)
->int_value(1)
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + i: INTEGER
      Number of queries: 0
      Number of commands: 1
        + set_i(INTEGER, INTEGER)
  Routine currently being implemented: {A}.set_i
  Assignment being specified: i := ((1 + ?) * nil)
->int_value(2)
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + i: INTEGER
      Number of queries: 0
      Number of commands: 1
        + set_i(INTEGER, INTEGER)
  Routine currently being implemented: {A}.set_i
  Assignment being specified: i := ((1 + 2) * ?)
->int_value(3)
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + i: INTEGER
      Number of queries: 0
      Number of commands: 1
        + set_i(INTEGER, INTEGER)
->add_assignment_instruction("A","set_i","i")
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + i: INTEGER
      Number of queries: 0
      Number of commands: 1
        + set_i(INTEGER, INTEGER)
  Routine currently being implemented: {A}.set_i
  Assignment being specified: i := ?
->addition
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + i: INTEGER
      Number of queries: 0
      Number of commands: 1
        + set_i(INTEGER, INTEGER)
  Routine currently being implemented: {A}.set_i
  Assignment being specified: i := (? + nil)
->add_call_chain(<<"i">>)
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + i: INTEGER
      Number of queries: 0
      Number of commands: 1
        + set_i(INTEGER, INTEGER)
  Routine currently being implemented: {A}.set_i
  Assignment being specified: i := (i + ?)
->subtraction
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + i: INTEGER
      Number of queries: 0
      Number of commands: 1
        + set_i(INTEGER, INTEGER)
  Routine currently being implemented: {A}.set_i
  Assignment being specified: i := (i + (? - nil))
->add_call_chain(<<"x">>)
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + i: INTEGER
      Number of queries: 0
      Number of commands: 1
        + set_i(INTEGER, INTEGER)
  Routine currently being implemented: {A}.set_i
  Assignment being specified: i := (i + (x - ?))
->add_call_chain(<<"y">>)
  Status: OK.
  Number of classes being specified: 1
    A
      Number of attributes: 1
        + i: INTEGER
      Number of queries: 0
      Number of commands: 1
        + set_i(INTEGER, INTEGER)
->type_check
  class A is type-correct.
->generate_java_code
  class A {
    int i;
    void set_i(int x, int y) {
      i = ((1 + 2) * 3);
      i = (i + (x - y));
    }
  }
